addons/sale/models/sale.py

```diff
@@ -1620,6 +1620,61 @@ def product_id_change(self):
 
         return result
 
+    def _get_computed_price_unit(self, product=None):
+        ''' Helper to get the default price unit based on the product by taking care of the taxes
+        set on the product and the fiscal position.
+        :return: The price unit.
+        '''
+        self.ensure_one()
+
+        if not self.product_id:
+            return 0.0
+
+        company = self.order_id.company_id
+        currency = self.order_id.currency_id
+        company_currency = company.currency_id
+        fiscal_position = self.order_id.fiscal_position_id
+        is_refund_document = self.product_uom_qty < 0
+        order_date = self.order_id.date_order or fields.Date.context_today(self)
+
+        product_price_unit = self._get_display_price(product)
+        product_taxes = self.product_id.taxes_id.filtered(lambda tax: tax.company_id == company)
+
+        # Apply fiscal position.
+        if product_taxes and fiscal_position:
+            product_taxes_after_fp = fiscal_position.map_tax(product_taxes, partner=self.order_id.partner_id)
+
+            if set(product_taxes.ids) != set(product_taxes_after_fp.ids):
+                flattened_taxes_before_fp = product_taxes._origin.flatten_taxes_hierarchy()
+                if any(tax.price_include for tax in flattened_taxes_before_fp):
+                    taxes_res = flattened_taxes_before_fp.compute_all(
+                        product_price_unit,
+                        quantity=1.0,
+                        currency=company_currency,
+                        product=self.product_id,
+                        partner=self.order_id.partner_id,
+                        is_refund=is_refund_document,
+                    )
+                    product_price_unit = company_currency.round(taxes_res['total_excluded'])
```

Comment:
The rounding of `total_excluded` is already managed by `compute_all`.
The price_unit is not a monetary field and has his own precision (see `digits` on field)
So, `product_price_unit = taxes_res['total_excluded']`